tests/test_packet_callbacks.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import zigpy.types as t
+from .async_mock import MagicMock
+from .conftest import make_ieee
+
+
+def make_packet(src_address=None, **overrides):
+    """Build a packet."""
+    defaults = {
+        "src": src_address
+        or t.AddrModeAddress(addr_mode=t.AddrMode.NWK, address=0x1234),
+        "src_ep": 1,
+        "dst": t.AddrModeAddress(addr_mode=t.AddrMode.NWK, address=0x0000),
+        "dst_ep": 1,
+        "tsn": 123,
+        "profile_id": 0x0104,
+        "cluster_id": 0x0006,
+        "data": t.SerializableBytes(b"test"),
+        "lqi": 255,
+        "rssi": -30,
+    }
+    defaults.update(overrides)
+    return t.ZigbeePacket(**defaults)
```

Comment:
The ZigbeePacket type implements a `.replace(dst_ep=123)` method that can replace most uses of this function.